### PR TITLE
Convert /test references to SDKv2

### LINKED ISSUE #3985: Update /test to SDKv2

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Converts references in the `/test` directory to azure-sdk-for-go version 2.

**Which issue(s) this PR fixes**:

Fixes #3985

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mboersma** [comment]
/retitle [WIP] Convert /test references to SDKv2

There's more to change here, but I'm taking baby steps.

File: test/e2e/azure_securitygroups.go [hunk 1]
```diff
@@ -91,18 +91,17 @@ func AzureSecurityGroupsSpec(ctx context.Context, inputGetter func() AzureSecuri
 	settings, err := auth.GetSettingsFromEnvironment()
 	Expect(err).NotTo(HaveOccurred())
 	subscriptionID := settings.GetSubscriptionID()
-	auth, err := azureutil.GetAuthorizer(settings)
 	Expect(err).NotTo(HaveOccurred())
 
-	By("creating a subnets client")
-	subnetsClient := network.NewSubnetsClient(subscriptionID)
```
Comment (mboersma) [inline test/e2e/azure_securitygroups.go:98]: This `subnetsClient` wasn't actually being used.

**Comment by mboersma** [comment]
/test pull-cluster-api-provider-azure-e2e-optional

There is still `test/e2e/azure_privatecluster.go` to add, but I want see that we haven't already broken anything.

**Comment by mboersma** [comment]
Must have broken something in `GetAKSKubernetesVersion()` (although the AKS test passed locally). I'm investigating.

**Comment by mboersma** [comment]
/test pull-cluster-api-provider-azure-e2e-optional

For the private cluster test spec.

**Comment by mboersma** [comment]
Yay, AKS is passing now. I'm looking into the VMSS flex failure, then the private cluster spec.

**Comment by mboersma** [comment]
/retitle Convert /test references to SDKv2

I've removed the private cluster-specific changes from this PR so I can troubleshoot that separately (see #4056). These changes appear to work (on the private cluster test as well).

**Review by nojnhuh — COMMENTED** [review]
/lgtm

**Pull request merged**